src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NugetApiPackageManager.cs

```diff
@@ -0,0 +1,251 @@
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.Installer;
+using NuGet.Common;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using SemanticVersion = Microsoft.TemplateEngine.Abstractions.SemanticVersion;
+using NuGetSemanticVersion = NuGet.Versioning.SemanticVersion;
+using System;
+using System.Collections.Concurrent;
+using NuGet.Credentials;
+
+namespace Microsoft.TemplateEngine.Edge.Installers.NuGet
+{
+    internal class NuGetApiPackageManager : IDownloader, IUpdateChecker
+    {
+        private ILogger _nugetLogger = NullLogger.Instance;
+        internal const string PublicNuGetFeed = "https://api.nuget.org/v3/index.json";
+        private IEngineEnvironmentSettings _environmentSettings;
+
+        internal NuGetApiPackageManager(IEngineEnvironmentSettings settings)
+        {
+            IEnumerable<INuGetLogger> availableLoggers = settings.SettingsLoader.Components.OfType<INuGetLogger>();
+            if (availableLoggers.Any())
+            {
+                _nugetLogger = availableLoggers.First();
+            }
+            _environmentSettings = settings;
+        }
+
+        public bool CanCheckForUpdate(NuGetManagedTemplatesSource source)
+        {
+            return true;
+        }
+
+        public bool CanDownloadPackage(InstallRequest installRequest)
+        {
+            return true;
+        }
+
+        public async Task<DownloadResult> DownloadPackageAsync(InstallRequest installRequest, string downloadPath)
+        {
+            IEnumerable<string> sources = null;
+            if (installRequest.Details.ContainsKey(InstallRequest.NuGetSourcesKey))
+            {
+                sources = installRequest.Details[InstallRequest.NuGetSourcesKey].Split(InstallRequest.NuGetSourcesSeparator);
+            }
+            bool interactive = installRequest.Details.ContainsKey(InstallRequest.InteractiveModeKey);
+            DefaultCredentialServiceUtility.SetupDefaultCredentialService(_nugetLogger, !interactive);
```

Comment:
I think this call should be moved to CLI project, and each host is responsible for calling this and setting up NuGet credentials...